src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -389,6 +466,206 @@ ForeignConstraintFindDistKeys(HeapTuple pgConstraintTuple,
 }
 
 
+/*
+ * ErrorIfUnsupportedAlterAddDropFKeyBetweenReferecenceAndLocalTable runs checks related to ALTER
+ * ADD / DROP foreign key constraints between local tables and reference tables.
+ *
+ * If constraint is not NULL, then we perform checks for ADD command, otherwise we check for DROP.
```

Comment:
this can be confusing, I'd prefer passing an additional `AlterTableType alterTableType` argument instead of relying on `NULL`.

Reply:
makes sense